### DISCUSSION TITLE
C4 Model using Mermaid

### DISCUSSION THREAD

**Discussion by werberger** [GeriLife/companionship-care#202] [Q&A]
@brylie I have created some draft diagrams in Mermaid. Namely a Context, Container and two Component diagrams.

I have a number of questions but my main one is whether they're even along the lines of what you are thinking?
Either way would appreciate your thoughts before I create a PR.

Worth nothing that I'm not sure I'm sufficiently across the project (conceptually or technically) to build very meaningful diagrams. So the more specific/detailed you can be, the more likely I'll accurately make the changes you'd like.

And just checking, I should create an issue before making a PR anyway, right?

![image](https://user-images.githubusercontent.com/106753124/219369168-30c3d300-6339-4439-a0b0-5ee23d19eeee.png)

# Component Diagram
![image](https://user-images.githubusercontent.com/106753124/219369409-948bb134-38de-40a5-ab77-a803dfb9ea5f.png)

# Component Diagram (Web Application)
![image](https://user-images.githubusercontent.com/106753124/219370044-600dff03-5533-49c1-8448-5972aaa65ffb.png)

# Component Diagram (REST API)
![image](https://user-images.githubusercontent.com/106753124/219370143-4d8cd33a-79d6-4797-aa27-0b554a531a18.png)

**Comment by brylie** [comment]
This looks great! Let's try embedding the Mermaid source code directly in a Markdown file in our project documentation. E.g., create a new `architecture.md` file and use the GitHub mermaid codefence to render the diagrams.

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams

**Reply by werberger** [reply]
So glad. Plus I already did these diagrams using Mermaid in an architecture.md file :)

Can create a pull request but do you want me to create an issue first?

**Reply by brylie** [reply]
Great 🙂

Yes, ideally around 95% of pull requests would be linked to an issue, so go ahead and open the issue and then a linked PR. It's just to practice a good habit.
